src/main/java/com/aws/greengrass/deployment/CurrentDeploymentFinisher.java

```diff
@@ -0,0 +1,307 @@
+import static com.aws.greengrass.deployment.DeploymentService.DEPLOYMENT_ERROR_TYPES_KEY;
+import static com.aws.greengrass.deployment.DeploymentService.DEPLOYMENT_FAILURE_CAUSE_KEY;
+import static com.aws.greengrass.deployment.DeploymentService.DEPLOYMENT_ID_LOG_KEY_NAME;
+import static com.aws.greengrass.deployment.DeploymentService.GG_DEPLOYMENT_ID_LOG_KEY_NAME;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_MEMBERSHIP_TOPICS;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_TO_LAST_DEPLOYMENT_CONFIG_ARN_KEY;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_TO_LAST_DEPLOYMENT_TIMESTAMP_KEY;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_TO_LAST_DEPLOYMENT_TOPICS;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_TO_ROOT_COMPONENTS_GROUP_CONFIG_ARN;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_TO_ROOT_COMPONENTS_GROUP_NAME;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_TO_ROOT_COMPONENTS_TOPICS;
+import static com.aws.greengrass.deployment.DeploymentService.GROUP_TO_ROOT_COMPONENTS_VERSION_KEY;
+import static com.aws.greengrass.deployment.converter.DeploymentDocumentConverter.LOCAL_DEPLOYMENT_GROUP_NAME;
+import static com.aws.greengrass.deployment.model.DeploymentResult.DeploymentStatus.FAILED_ROLLBACK_NOT_REQUESTED;
+
+
+public class CurrentDeploymentFinisher {
+
+    private final Logger logger;
+    private final Deployment deployment;
+    private final DeploymentStatusKeeper deploymentStatusKeeper;
+    private final DeploymentDirectoryManager deploymentDirectoryManager;
+    private final Topics config;
+
+    @Inject
+    private final Kernel kernel;
+
+    /**
+     * Constructor for DeploymentFinisher.
+     *
+     * @param logger                     Logger instance
+     * @param deployment                 Deployment instance
+     * @param deploymentStatusKeeper     {@link DeploymentStatusKeeper}
+     * @param deploymentDirectoryManager {@link DeploymentDirectoryManager}
+     * @param kernel                     {@link Kernel}
+     * @param config                     root Configuration topic for Deployment service
+     */
+    public CurrentDeploymentFinisher(Logger logger, Deployment deployment,
+                                     DeploymentStatusKeeper deploymentStatusKeeper,
+                                     DeploymentDirectoryManager deploymentDirectoryManager, Topics config,
+                                     Kernel kernel) {
+        this.logger = logger;
+        this.deployment = deployment;
+        this.deploymentStatusKeeper = deploymentStatusKeeper;
+        this.deploymentDirectoryManager = deploymentDirectoryManager;
+        this.config = config;
+        this.kernel = kernel;
+    }
+
+    void finishCurrentDeployment(DeploymentResult result, DeploymentTask deploymentTask) throws InterruptedException {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 12. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.